lib/msf/core/rpc/v10/rpc_session.rb

```diff
@@ -163,6 +163,54 @@ def rpc_shell_upgrade( sid, lhost, lport)
     { "result" => "success" }
   end
 
+  # Writes to an interactive session (i.e. DB and SMB sessions). Note that you will to manually add a newline at the
+  # end of your input so the system will process it.
+  # You may want to use #rpc_interactive_shell_read to retrieve the output.
+  #
+  # @param [Integer] sid Session ID.
+  # @param [String] data The data to write.
+  #
+  # @return [Hash] A hash indicating the action was successful. It contains the following key:
+  # * 'result' [String] A message that says 'success'
+  #
+  # @example Here's how you would use this from the client:
+  # rpc.call('session.interactive_shell_write', 2, "DATA")
+  # => {"result"=>"success"}
+  def rpc_interactive_shell_write(sid, data)
+    # Session type is mandatory argument but `_valid_session` doesn't actually use it for interactive session types,
+    # and we don't know exact type of the session, so we just pass empty string
+    session = _valid_session(sid, '')
```

Comment:
This feels like a code smell; the method pasing an empty string to bypass the validation logic - as well as `rpc_interactive_shell_write` being a very generic name - but it actually only applies to specific session types

I wonder if we should mark the old meterpreter method as deprecated entirely and have everything go through this new method instead. Or, continue down the path of namespacing the RPC methods? 🤔

Reply:
Good point, let's discuss. I think if we improve `_valid_session` method to actually validate things it would be better

Reply:
Or take in the expected session type as a parameter, as it seems to only be used as a safe guard check.

Reply:
> `rpc_interactive_shell_write` being a very generic name - but it actually only applies to specific session types

Also I wonder if `interactive` is taking a good name, and might be an issue when we introduce a new session type that can be interacted with but in a different capacity

Reply:
Open to suggestions for better name, I struggle to find anything